documentation/public/user-guide/npm-package/using-queries.md

```diff
@@ -0,0 +1,81 @@
+# Using Queries
+
+It's often more convenient to use the declarative `Query` API to traverse the CST, as they allow you to express your intent more concisely and can largely replace the need for both internal (cursor), and external (visitor) iterator patterns.
+
+The [query language](./../query-language.md) is based on pattern matching, and the execution semantics are closer to unification than to regular expression matching. A query returns all possible matches, not just the longest/shortest/first/last match.
+
+If not specified otherwise, let's assume we already parsed a Solidity source and have a `cursor` pointing to the root node of the CST (created with `createTreeCursor`, see [Using the Cursor](./using-the-cursor.md)).
+
+## Creating and executing queries
+
+You can create a `Query` struct using `Query.parse`, which accepts a string value. These can be then used by `Cursor.query` to execute it. You can pass multiple queries to a cursor to and efficiently traverse the tree looking for matches.
```

Comment:
nit: `object` vs `struct` in JS:

```suggestion
You can create a `Query` object using `Query.parse`, which accepts a string value. These can be then used by `Cursor.query` to execute it. You can pass multiple queries to a cursor to and efficiently traverse the tree looking for matches.
```

Reply:
Good catch! Also added a variant of the note to both of the chapters (rs/ts).